src/front-door/azext_front_door/_params.py

```diff
@@ -50,11 +59,14 @@ def load_arguments(self, _):
         c.argument('frontend_host_name', help='Domain name of the frontend endpoint.')
 
     with self.argument_context('network front-door', arg_group='HTTPS') as c:
-        c.argument('certificate_source', arg_type=get_enum_type(FrontDoorCertificateSource), help='Certificate source to enable HTTPS. Allowed values: FrontDoor, AzureKeyVault. For Azure Key Vault, right permissions need to be set for Front Door to to access the Key vault. Learn more at https://aka.ms/FrontDoorCustomDomain.')
+        c.argument('certificate_source', arg_type=get_enum_type(FrontDoorCertificateSource), help='Certificate source to enable HTTPS. Allowed values: FrontDoor, AzureKeyVault. For Azure Key Vault, right permissions need to be set for Front Door to access the Key vault. Learn more at https://aka.ms/FrontDoorCustomDomain.')
         c.argument('secret_name', help='The name of the Key Vault secret representing the full certificate PFX')
         c.argument('secret_version', help='The version of the Key Vault secret representing the full certificate PFX')
         c.argument('vault_id', help='The resource id of the Key Vault containing the SSL certificate')
 
+    with self.argument_context('network front-door', arg_group='BackendPools Settings') as c:
+        c.argument('enforce_certificate_name_check', arg_type=get_enum_type(EnforceCertificateNameCheckEnabledState), help='Whether to enforce certificate name check on HTTPS requests to all backend pools. No effect on non-HTTPS requests.')
```

Comment:
Are the values here just true/false or enabled/disabled?  If so, I'd recommend using `arg_type=get_three_state_flag(...)`

Reply:
Fixed